owfs.js

```diff
@@ -27,46 +27,45 @@ module.exports = function(RED) {
+        node.on("input", function(msg) {
+            var host = msg.host  ||  node.host;
+            var port = msg.port  ||  node.port;
+            //node.log("owserver " + host + ":" + port);
+            var client = new owfs.Client(host, port);
+            var paths = node.paths;
+            if (msg.topic) {
+                paths = [msg.topic];
+            }
 
-                if (paths && paths.length > 0) {
-                    // Query owfs for each path, one at a time
-                    async.eachSeries(paths, function(path, callback) {
-                        client.read(path, function(error, result) {
-                            if (!error) {
-                                if (result.match(/^\-?\d+\.\d+$/)) {
-                                    msg.payload = parseFloat(result);
-                                } else if (result.match(/^\-?\d+$/)) {
-                                    msg.payload = parseInt(result);
-                                } else {
-                                    msg.payload = result;
-                                }
-                                msg.topic = path;
-                                msg.timestamp = Date.now();
-                                node.send(msg);
+            if (paths && paths.length > 0) {
+                // Query owfs for each path, one at a time
+                async.eachSeries(paths, function(path, callback) {
+                    client.read(path, function(error, result) {
+                        if (!error) {
+                            if (result.match(/^\-?\d+\.\d+$/)) {
+                                msg.payload = parseFloat(result);
+                            } else if (result.match(/^\-?\d+$/)) {
+                                msg.payload = parseInt(result);
+                            } else {
+                                msg.payload = result;
+                            }
+                            msg.topic = path;
+                            msg.timestamp = Date.now();
+                            node.send(msg);
+                        } else {
+                            if ('msg' in error) {
+                                node.error(error.msg, msg);
                             } else {
-                                if ('msg' in error) {
-                                    node.error(error.msg, msg);
-                                } else {
-                                    node.error(error, msg);
-                                }
+                                node.error(error, msg);
                             }
-                            callback();
-                        });
+                        }
+                        callback();
                     });
-                } else {
-                    node.warn("no owfs paths configured and msg.topic is empty");
-                }
-            });
-        } else {
-            node.error("missing host and port configuration for owfs");
```

Comment:
Good point.
At design time it is ok as it shows the missing configuration message so it stops the user scanning for devices.
Will check out what happens at run time.